docs/docs/reference/contextual/givens.md

```diff
@@ -90,6 +90,21 @@ transparent inline given mkAnnotations[A, T] as Annotations[A, T] = ${
 ```
 Since `mkAnnotations` is `transparent`, the type of an application is the type of its right hand side, which can be a proper subtype of the declared result type `Annotations[A, T]`.
 
+## Pattern-Bound Given Instances
+
+Given instances can also appear as pattern bound-variables. Example:
+
+```scala
+for given Context <- applicationContexts do
+
+pair match
+  case (ctx as given Context, y) => ...
```

Comment:
We should have tests with nested given patterns.

Reply:
you can't make `x` a given here. for `x` and `y` it would simply be:
```
case (x as given Y, y as given Z)
```
We already have a test like this.

Reply:
This seems to be a core limitation of the syntax. 

The given statement syntax is
```scala
given X
given x as X
```
What if we used the same in patterns?
```scala
case given X =>
case given x as X =>
```

Then we could extend it to

```scala
case given x @ X(y, z) =>
```